src/Device/Driver/AR62xx.cpp

```diff
@@ -0,0 +1,573 @@
+  else if (channel <= 65)
+    frequency_id += channel / 5 - 2;
+  else if (channel <= 90)
+    frequency_id += channel / 5 - 3;
+
+  return (frequency_id);
+}
+
+/**
+ * @brief bitwise cyclic redundancy check - algorithm
+ * 
+ * @param data data which should be checked
+ * @param len length of data
+ * @return uint16_t checksum
+ */
+static uint16_t CRCBitwise(uint8_t *data, size_t len)
+{
+  uint16_t crc = 0x0000;
+  for (size_t j = len; j > 0; j--)
+  {
+    crc ^= (uint16_t)(*data++) << 8;
+    for (int i = 0; i < 8; i++)
+    {
+      if (crc & 0x8000)
+        crc = (crc << 1) ^ 0x8005;
+      else
+        crc <<= 1;
+    }
+  }
+  return (crc);
+}
+
+/**
+   * @brief generates a command to set a station in the AR62xx
+   * 
+   * @param command command-output to set the station
+   * @param station_id the id of the station which should be set
+   *        1 = active station of AR62xx
+   *        0 = passive station of AR62xx
+   * @param frequency frequency in kHz to set
+   * @param station name of the station to set
+   *        (unused)
+   * @return int command-length
+   */
+int AR62xxDevice::GenerateSetStationCommand(uint8_t *command, int active_passive, RadioFrequency frequency, const TCHAR *station)
+{
+  /** length of the command */
+  unsigned int command_length = 0;
+
+  /** convert the active-frequency to an AR62xx-ID */
+  uint16_t active_frequency_idx = FrequencyToAR62xxId(active_frequency);
+
+  /** convert the passive-frequency to an AR62xx-ID */
+  uint16_t passive_frequency_idx = FrequencyToAR62xxId(passive_frequency);
+
+  /** command-start */
+  command[command_length++] = 0xA5;
+
+  /** add prot-ID */
+  command[command_length++] = 0x14;
+  command[command_length++] = 5;
+
+  /** set active station */
+  if (active_passive == ACTIVE_STATION)
+    active_frequency_idx = FrequencyToAR62xxId(frequency);
+
+  /** set passive station */
+  else
+    passive_frequency_idx = FrequencyToAR62xxId(frequency);
+
+  /** setting frequencies */
+  command[command_length++] = 22;
+  command[command_length++] = GetNthByte(active_frequency_idx, 1);
+  command[command_length++] = GetNthByte(active_frequency_idx, 0);
+  command[command_length++] = GetNthByte(passive_frequency_idx, 1);
+  command[command_length++] = GetNthByte(passive_frequency_idx, 0);
+
+  /** Creating the binary value */
+  uint16_t command_crc = CRCBitwise(command, command_length);
+  command[command_length++] = GetNthByte(command_crc, 1);
+  command[command_length++] = GetNthByte(command_crc, 0);
+  return command_length;
+}
+
+/**
+   * @brief parses the received data
+   * this is needed to save the active and passive station
+   * 
+   * @param string recived data
+   * @param len length of data
+   * @return true if data could be parsed
+   * @return false if data could not be parsed
+   */
+bool AR62xxDevice::ParseReceivedData(const char *string, size_t len)
+{
+  /** check if there is data */
+  if (string == NULL || len == 0)
```

Comment:
Yet another `NULL` check. Why?
And why check for `len == 0`? Serious question.

Reply:
removed checks as these are not needed